### ISSUE TITLE
react-hooks/exhaustive-deps: missing dependency not reported when the callback destructures a whole object and the dep array lists only a member path

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25621]
## Summary

`react-hooks/exhaustive-deps` fails to flag a missing dependency when a hook callback destructures a whole object (`const {x} = obj`) while the dependency array lists only a member path (`[obj.x]`). The whole `obj` is the real dependency, but oxlint reports nothing; `eslint-plugin-react-hooks` (on both ESLint 7 and ESLint 9) flags `missing dependency: 'context'`. The simpler member-read form is flagged correctly, so the gap is specifically in reconciling a whole-object read (via destructuring) against a member-path entry in the dep array. On one real-world file this defect class hides 17 exhaustive-deps findings that the real plugin reports.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0

## Reproduction

`repro.tsx`:

```tsx
import React from 'react'

export function useThing(context: {paramsLoader: () => void}) {
	// body destructures the WHOLE `context`; dep array lists only a member path
	const cb = React.useCallback(() => {
		const {paramsLoader} = context
		paramsLoader()
	}, [context.paramsLoader])
	return cb
}
```

`control.tsx` (correctly flagged — isolates the gap):

```tsx
import React from 'react'

export function useControl(source: {isAllLoaded: boolean}) {
	const cb = React.useCallback(() => source.isAllLoaded, [])
	return cb
}
```

`.oxlintrc.json`:

```json
{
	"plugins": ["react"],
	"categories": {"correctness": "off"},
	"settings": {"react": {"version": "18.3"}},
	"rules": {"react-hooks/exhaustive-deps": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config repro.tsx control.tsx
```

## Expected

A finding on `repro.tsx` equivalent to eslint-plugin-react-hooks:

```
React Hook React.useCallback has a missing dependency: 'context'. Either include it or remove the dependency array.
```

Rationale: destructuring `const {paramsLoader} = context` reads the whole `context` binding, so `context` (not `context.paramsLoader`) must appear in the dependency array. Listing `[context.paramsLoader]` does not satisfy the whole-object read.

## Actual

oxlint 1.78.0 output:

```
control.tsx:4:37: error react-hooks(exhaustive-deps): React Hook useCallback has a missing dependency: 'source.isAllLoaded' help: Either include it or remove the dependency array.
```

`repro.tsx` produces zero diagnostics (exit 0 when linted alone). Adding `/* eslint-disable react-hooks/exhaustive-deps */` to it and running with `--report-unused-disable-directives` reports the directive as "Unused eslint-disable directive (no problems were reported)", masking a genuine bug.

## Cross-checked

| Engine | repro.tsx | control.tsx |
| --- | --- | --- |
| ESLint 7.23.0 + eslint-plugin-react-hooks@4.2.0 (@typescript-eslint/parser@6.21.0) | 8:5 missing dependency: `'context'` | 4:57 missing dependency: `'source.isAllLoaded'` |
| ESLint 9.39.4 flat config + eslint-plugin-react-hooks@6.1.1 (typescript-eslint@8.62.1) | 8:5 missing dependency: `'context'` | 4:57 missing dependency: `'source.isAllLoaded'` |
| oxlint 1.78.0 | no diagnostics | 4:37 missing dependency: `'source.isAllLoaded'` |
